File: 240327/softeer-6268.py
number = []
number.append([1, 1, 1, 0, 1, 1, 1]) # 0
number.append([0, 0, 1, 0, 0, 0, 1]) # 1
number.append([1, 0, 1, 1, 1, 1, 0]) # 2
number.append([1, 0, 1, 1, 0, 1, 1]) # 3
number.append([0, 1, 1, 1, 0, 0, 1]) # 4
number.append([1, 1, 0, 1, 0, 1, 1]) # 5
number.append([1, 1, 0, 1, 1, 1, 1]) # 6
number.append([1, 1, 1, 0, 0, 0, 1]) # 7
number.append([1, 1, 1, 1, 1, 1, 1]) # 8
number.append([1, 1, 1, 1, 0, 1, 1]) # 9

T = int(input())
for _ in range(T):
    A, B = map(str, input().split())

    answer = 0
    if len(A) == len(B):
        for position in range(0, len(A)):
            for i in range(7):
                if number[int(A[position])][i] != number[int(B[position])][i]:
                    answer += 1
                    
    elif len(A) > len(B):
        for position in range(0, len(A) - len(B)):
            for i in range(7):
                if number[int(A[position])][i] == 1:
                    answer += 1
        A = A[len(A) - len(B):]
        for position in range(len(B)):
            for i in range(7):
                if number[int(A[position])][i] != number[int(B[position])][i]:
                    answer += 1

    elif len(A) < len(B):
        for position in range(0, len(B) - len(A)):
            for i in range(7):
                if number[int(B[position])][i] == 1:
                    answer += 1
        B = B[len(B) - len(A):]
        for position in range(len(A)):
            for i in range(7):
                if number[int(A[position])][i] != number[int(B[position])][i]:
                    answer += 1

    print(answer)

File: 240327/softeer-6268.cpp
#include<iostream>
#include <vector>
#include <string>

using namespace std;

int main(int argc, char** argv){
    vector<vector<int>> number(10);
    number[0] = {1, 1, 1, 0, 1, 1, 1};
    number[1] = {0, 0, 1, 0, 0, 0, 1};
    number[2] = {1, 0, 1, 1, 1, 1, 0};
    number[3] = {1, 0, 1, 1, 0, 1, 1};
    number[4] = {0, 1, 1, 1, 0, 0, 1};
    number[5] = {1, 1, 0, 1, 0, 1, 1};
    number[6] = {1, 1, 0, 1, 1, 1, 1};
    number[7] = {1, 1, 1, 0, 0, 0, 1};
    number[8] = {1, 1, 1, 1, 1, 1, 1};
    number[9] = {1, 1, 1, 1, 0, 1, 1};

    int T;
    cin >> T;
    for(int t=0; t<T; t++){
        string A, B;
        cin >> A >> B;

        int answer = 0;
        if(A.length() == B.length()){
            for(int position=0; position<A.length(); position++){
                for(int i=0; i<7; i++){
                    if(number[A[position] - '0'][i] != number[B[position] - '0'][i]){
                        answer += 1;
                    }
                }
            }
        } else if(A.length() > B.length()){
            for(int position=0; position<(A.length() - B.length()); position++){
                for(int i=0; i<7; i++){
                    if(number[A[position] - '0'][i] == 1){
                        answer += 1;
                    }
                }
            }
            A = A.substr(A.length() - B.length());
            for(int position=0; position<B.length(); position++){
                for(int i=0; i<7; i++){
                    if(number[A[position] - '0'][i] != number[B[position] - '0'][i]){
                        answer += 1;
                    }
                }
            }
        } else if(A.length() < B.length()){
            for(int position=0; position<(B.length() - A.length()); position++){
                for(int i=0; i<7; i++){
                    if(number[B[position] - '0'][i] == 1){
                        answer += 1;
                    }
                }
            }
            B = B.substr(B.length() - A.length());
            for(int position=0; position<A.length(); position++){
                for(int i=0; i<7; i++){
                    if(number[A[position] - '0'][i] != number[B[position] - '0'][i]){
                        answer += 1;
                    }
                }
            }
        }
        cout << answer << endl;
    }
   return 0;
}